mono/mini/exceptions-amd64.c

```diff
@@ -894,6 +898,10 @@ mono_arch_handle_altstack_exception (void *sigctx, MONO_SIG_HANDLER_INFO_TYPE *s
 	gpointer *sp;
 	int frame_size;
 	MonoContext *copied_ctx;
+	gboolean native = FALSE;
+
+	if (!mono_is_addr_implicit_null_check (fault_addr))
+		native = TRUE;
```

Comment:
`mono_is_addr_implicit_null_check` returning `false` doesn't imply that the fault is coming from C code. it can also be buggy JIT code.

Reply:
Note that the semantics of these checks were not changed, just the checks + native crash handler moved into the code which is no longer on altstack.